### PR TITLE
Array avg

### LINKED ISSUE #468: Fix arrayAvg method
There is currently a method called `arrayAvg` that lives in `/lib/arrayAvg.js`.
It is incomplete and needs to be fixed!

### Acceptance Criteria:
1. Running `yarn test <methodName>` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by joshuafbarker**
Closes #468 

- [x] Running `yarn lint` does not trigger any linter errors
- [x] The test of the method you have fixed is passing
- [x] You have written a new skeleton method for someone else to work on!
- [x] You have written tests to accompany your skeleton method

### CODE REVIEW

File: lib/arrayOrder.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Takes an array, and orders the numbers from least to greatest.
+ * 
+ * @param {Array} An array of numbers.
+ *
+ * @return {Array} An array of numbers ordered from least to greatest.
+ * 
+ */
+function arrayOrder(arr) {
```
Comment (ktilcu) [inline lib/arrayOrder.js]: Can we call this `sort`?

Comment (joshuafbarker) [reply lib/arrayOrder.js]: sortArray or arraySort?

I'm cool with either.

Comment (ktilcu) [reply lib/arrayOrder.js]: I was think just `sort`. Is there anything else we can sort?

Comment (joshuafbarker) [reply lib/arrayOrder.js]: Cool. I will go ahead and make those updates.

Comment (ktilcu) [reply lib/arrayOrder.js]: Awesome! This is totally up for discussion too. If you had a different idea feel free to tell me to back off. Thanks and Happy Hacktoberfest 🎃 !

Comment (joshuafbarker) [reply lib/arrayOrder.js]: I'm stuck on the test. Not sure how I would test with accepting a function, because the result would depend on the sorting function that was entered.

Sorry, I am new to writing tests.

Comment (joshuafbarker) [reply lib/arrayOrder.js]: Never mind, I think I know what to do.

**Comment by joshuafbarker** [comment]
I went ahead and made the requested changes @ktilcu

File: test/sortBy.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { sortBy } = require("../lib");
+
+describe("sortBy", () => {
+  test("ascending order from [3, 1, 2] should be [1, 2, 3]", () => {
+    expect(sortBy((a, b) => a > b, [3, 1, 2])).toBe([1, 2, 3]);
```
Comment (ktilcu) [inline test/sortBy.test.js]: likely need to use `toEqual` here otherwise it checks for the same reference variable and not the equality of the values.

File: test/sortBy.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  });
+
+  test("descending order from [-1, -3, 2, 1, 5, 0] should be [5, 2, 1, 0, -1, -3]", () => {
+    expect(sortBy((a, b) => a < b, [-1, -3, 2, 1, 5, 0])).toBe([5, 2, 1, 0, -1, -3]);
+  });
+
+  test("ascending order from [{x: 3, y: 8}, {x: 1, y: 7}] should be [{x: 1, y: 7}, {x: 3, y: 8}]", () => {
+    expect(sortBy((a, b) => a.x > b.x, [{x: 3, y: 8}, {x: 1, y: 7}])).toBe([{x: 1, y: 7}, {x: 3, y: 8}]);
+  });
+});
```
Comment (ktilcu) [inline test/sortBy.test.js]: Great work on the testing! I would make sure you make test for edge cases. missing function, missing list, empty list,  wrong types, etc...

**Comment by ktilcu** [comment]
@joshuafbarker Got it. Reviewed!

File: test/sortBy.test.js (hunk 3 above)
Comment (joshuafbarker) [reply test/sortBy.test.js]: Ok.. so in one of those instances, would it be something like..

`
test("ascending order from [{x: 3, y: 8}, {x: 1, y: 7}] should be [{x: 1, y: 7}, {x: 3, y: 8}]", () => {
    expect(sortBy((a, b) => a.x > b.x)).toBeFalsy();
});
`

Comment (ktilcu) [reply test/sortBy.test.js]: There is also a `toThrow` method on the expect object. `expect(...).toThrow()`

**Comment by joshuafbarker** [comment]
@ktilcu 

Added more tests. If there are any more cases you can think of let me know, and I will add!
